packages/snaps-utils/src/snaps.test.ts

```diff
@@ -52,14 +52,14 @@ describe('assertIsValidSnapId', () => {
         // TODO: Either fix this lint violation or explain why it's necessary to
         //  ignore.
         // eslint-disable-next-line @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-base-to-string
-        `Invalid snap ID: Expected the value to satisfy a union of \`intersection | string\`, but received: ${value}.`,
+        `Invalid snap ID: Expected the value to satisfy a union of \`intersection | Base Snap Id\`, but received: ${value}.`,
```

Comment:
With `selectiveUnion` we could probably improve this message more.

Reply:
Let me give it a try